grpc/src/client/load_balancing/child_manager.rs

```diff
@@ -23,27 +23,39 @@
 // policy in use.  Complete tests must be written before it can be used in
 // production.  Also, support for the work scheduler is missing.
 
+use std::{fmt::Debug, sync::atomic::{AtomicUsize, Ordering}};
 use std::collections::HashSet;
+use std::fmt::Display;
 use std::sync::Mutex;
 use std::{collections::HashMap, error::Error, hash::Hash, mem, sync::Arc};
 
 use crate::client::load_balancing::{
-    ChannelController, LbConfig, LbPolicy, LbPolicyBuilder, LbPolicyOptions, LbState,
-    WeakSubchannel, WorkScheduler,
+    ChannelController, ExternalSubchannel, Failing, LbConfig, LbPolicy, LbPolicyBuilder,
+    LbPolicyOptions, LbState, ParsedJsonLbConfig, PickResult, Picker, QueuingPicker,
+    Subchannel, SubchannelState, WeakSubchannel, WorkScheduler, GLOBAL_LB_REGISTRY, ConnectivityState,
 };
 use crate::client::name_resolution::{Address, ResolverUpdate};
+use crate::service::{Message, Request, Response, Service};
 
-use super::{Subchannel, SubchannelState};
+use tonic::{metadata::MetadataMap, Status};
+
+use tokio::sync::{mpsc, watch, Notify};
+use tokio::task::{AbortHandle, JoinHandle};
 
 // An LbPolicy implementation that manages multiple children.
 pub struct ChildManager<T> {
     subchannel_child_map: HashMap<WeakSubchannel, usize>,
     children: Vec<Child<T>>,
     update_sharder: Box<dyn ResolverUpdateSharder<T>>,
     pending_work: Arc<Mutex<HashSet<usize>>>,
+    updated: bool, // true if a child has updated its state since the last call to has_updated.
+    sent_connecting_state: bool,
+    aggregated_state: ConnectivityState,
+    last_ready_pickers: Vec<Arc<dyn Picker>>,
+
```

Comment:
Please remove extraneous blank lines.  Blank lines should be intentionally visually separating things.